person.rb

```diff
@@ -0,0 +1,26 @@
+# class person
+class Person
+  # we set the getter and setter for these elements - id, name and age
+  attr_accessor :id, :name, :age
+
+  # constructor
+  def initialize(id, age, name = 'unknown', parent_permission: true)
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  # public methods
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  # private methods
```

Comment:
Beside setting up  person, student and teacher classes most of the project's requirements are missing. kindly read the project requirements carefully the work on them.
You are supposed to :

- [ ] Create a class Nameable.

- [ ] Implement a method called correct_name that will raise a NotImplementedError.

- [ ] Turn your Person class to Nameable

- [ ] Make sure that your Person class inherits from Nameable

- [ ] Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

- [ ] Prepare base Decorator

- [ ] Make sure that it inherits from Nameable.

- [ ] In the constructor assign a nameable object from params to an instance variable.

- [ ] Implement the correct_name method that returns the result of the correct_name method of the @nameable.

- [ ] Prepare CapitalizeDecorator and TrimmerDecorator

For the CapitalizeDecorator:

- [ ] Create a class that inherits from the base Decorator class.

- [ ] Implement a method correct_name that capitalizes the output of @nameable.correct_name.

For the TrimmerDecorator:

- [ ] Create a class that inherits from the base Decorator class.

- [ ] Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.